esp-ieee802154/src/compat/mod.rs

```diff
@@ -0,0 +1,130 @@
+use core::{ffi::VaListImpl, fmt::Write};
+
+use log::info;
+
+use self::str_buf::StrBuf;
+
+mod str_buf;
+
+#[no_mangle]
+pub unsafe extern "C" fn phy_printf(format: *const u8, args: ...) {
+    syslog(format, args);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rtc_printf(format: *const u8, args: ...) {
+    syslog(format, args);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn coexist_printf(format: *const u8, args: ...) {
+    syslog(format, args);
+}
+
+pub unsafe extern "C" fn syslog(format: *const u8, args: VaListImpl) {
+    let mut buf = [0u8; 512];
+    vsnprintf(&mut buf as *mut u8, 511, format, args);
+    let res_str = StrBuf::from(&buf as *const u8);
+    info!("{}", res_str.as_str_ref());
+}
+
+pub(crate) unsafe fn vsnprintf(
+    dst: *mut u8,
+    _n: u32,
+    format: *const u8,
+    mut args: VaListImpl,
+) -> i32 {
+    let fmt_str_ptr = format;
+
+    let mut res_str = StrBuf::new();
+
+    let strbuf = StrBuf::from(fmt_str_ptr);
+    let s = strbuf.as_str_ref();
+
+    let mut format_char = ' ';
+    let mut is_long = false;
+    let mut found = false;
+    for c in s.chars() {
```

Comment:
Is this `ascii` or `utf-8`? If `ascii` then better to use `bytes` instead of `chars`.